cmd/temporalite/main.go

```diff
@@ -237,6 +231,17 @@ func buildCLI() *cli.App {
 					}
 				}
 
+				interruptChan := make(chan interface{}, 1)
+				go func() {
+					if doneChan := c.Done(); doneChan != nil {
+						s := <-doneChan
+						interruptChan <- s
+					} else {
+						s := <-temporal.InterruptCh()
+						interruptChan <- s
+					}
+				}()
```

Comment:
```suggestion
				interruptChan := make(chan interface{})
				go func() {
					defer close(interruptChan)
					select {
					case <-c.Done():
					case <-temporal.IntteruptCh():
					}
				}()
```
Would miss interrupt if done were set w/ if/else (if `c.Done()` is `nil`, it just never completes here). Unless I misunderstand this bit of code.

Reply:
No I think this works as expected; ^C stops the server when invoked via `go run ./cmd/temporalite`, and if I add logging in the if/else code paths then `temporal.InterruptCh` is only used when Temporalite is not invoked with a cancelable context.

Reply:
Gotcha, I did misunderstand, you don't want to support signal-based interrupt for context-based runs.